R/zchunk_LA119.solar_china.R

```diff
@@ -0,0 +1,76 @@
+#' module_gcam.china_LA119.Solar
+#'
+#' Compute capacity factors for solar capacity factors by province.
+#'
+#' @param command API command to execute
+#' @param ... other optional parameters, depending on command
+#' @return Depends on \code{command}: either a vector of required inputs,
+#' a vector of output names, or (if \code{command} is "MAKE") all
+#' the generated outputs: \code{L119.CapFacScaler_PV_province}, \code{L119.CapFacScaler_CSP_province}. The corresponding file in the
+#' original data system was \code{LA114.Solar.R} (gcam-china level1).
+#' @details Computes solar capacity factor scalars for PV and CSP technologies by province, divided by the national average.
+#' @importFrom assertthat assert_that
+#' @importFrom dplyr filter mutate select
+#' @importFrom tidyr gather spread
+#' @author Liu July 2018
+module_gcam.china_LA119.Solar <- function(command, ...) {
+  if(command == driver.DECLARE_INPUTS) {
+    return(c(FILE = "gcam-china/solar_csp_pv_capacityfactor",
+             FILE = "gcam-china/province_names_mappings"))
+  } else if(command == driver.DECLARE_OUTPUTS) {
+    return(c("L119.CapFacScaler_PV_province",
+             "L119.CapFacScaler_CSP_province"))
+  } else if(command == driver.MAKE) {
+
+    fuel  <- CSP  <- sector <- scaler <-
+      province.nam <- province <- Centralized.PV <- NULL     # silence package check.
+
+    all_data <- list(...)[[1]]
+    # -----------------------------------------------------------------------------
+    # 1.Load required inputs
+    province_names_mappings     <- get_data(all_data, "gcam-china/province_names_mappings")
+    solar_csp_pv_capacityfactor <- get_data(all_data, "gcam-china/solar_csp_pv_capacityfactor")
+
+    # -----------------------------------------------------------------------------
+    # 2.perform computations
```

Comment:
# Create scalers to scale capacity factors read in the assumptions file.
# These scalers will then be used to create capacity factors by province.
# The idea is to vary capacity factors for solar technologies by province depending on the varying solar irradiance by province.

 # Create scalers by province by dividing capacity factor by the average